### DISCUSSION TITLE
chronycontrol wipes configuration on upgrade

### DISCUSSION THREAD

**Discussion by miquecg** [Homebrew/discussions#3968] [Casks]
Hi,

Today after running `brew upgrade` a new ChronyControl version came up and the previous configuration sitting on `/etc/chrony.d` was [removed](https://github.com/Homebrew/homebrew-cask/blob/HEAD/Casks/chronycontrol.rb#L21-L22).

Are there any mechanisms in place to keep own customizations from being overwritten? Should the cask code be updated to take this into account?

Thanks!

**Comment by carlocab** [comment]
We could probably move the `/etc/chrony.d` line to the `zap` stanza instead.

**Reply by miquecg** [reply]
Does it mean it would make a copy somewhere that I can grab and put back in place or is it something else?

**Reply by p-linnane** [reply]
Moving it from `uninstall` to `zap` would mean that it would only be removed if you specifically requsted it by passing `--zap` to `brew uninstall`. Otherwise it would remain and persist through upgrades.

**Reply by miquecg** [reply]
Great, that would work. I'm gonna request the change.

Thanks!

**Comment by miquecg** [comment]
Well, today it happened again despite moving `/etc/chrony.d` to `zap` stanza.

What I saw after upgrading was that chronycontrol prompted for installation on reboot as if it was first time install. In fact it overwrites `/etc/chrony.d`.

It looks like it needs to be solved upstream, doesn't it?

**Reply by p-linnane** [reply]
Yes, upstream will need to address that.

**Reply by miquecg** [reply]
Just to be sure about how Homebrew proceeds to do upgrades. Does it actually uninstall and then install the software again? That would mean the `zap` stanza will not be actually executed unless the user manually tell Homebrew to do so?

According to chronycontrol developer, it will not overwrite configuration directory if it exists.

**Reply by SMillerDev** [reply]
Yes, that is correct

**Reply by miquecg** [reply]
I've just manually uninstalled, followed by installation and reboot. Application started on boot as expected but this time folder configuration changes were still in place.

My guess is the upgrade caused the app to be uninstalled using old formula and then installed with the latest one. Not sure if that's even possible but kind of makes sense.
